### PR TITLE
Rewrite querying Vespa + bonus document gc options and tuning

### PR DESCRIPTION
**Pull request by jobergum**

### CODE REVIEW

File: documentation/performance/container-tuning.html [hunk 1]
```diff
@@ -17,157 +18,60 @@ <h2 id="container-worker-threads">Container worker threads</h2>
...
-</p><p>
-Set JVM parameters for Java services in <em>services.xml</em>.
-When modified, <em>vespa-deploy prepare</em>, <em>vespa-deploy activate</em> and restart the relevant services.
-Example: Run two containers with concurrent garbage collection:
+Use <em>jvm-gc-options</em> for controlling GC related parameters and <em>jvm-options</em> for tuning other parameters. See <a href="../reference/services-container.html#nodes">reference documentation</a>. 
+Example: Running with CMS concurrent garbage collection and using NewRatio = 1 (equal size of old and new generation) and enabling verbose GC logging (Logged to stdout to vespa.log file).
 <pre>
 &lt;container id="default" version="1.0"&gt;
-  &lt;nodes jvmargs="-XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:+PrintGCDetails -XX:+PrintGCTimeStamps -XX:+CMSIncrementalPacing"&gt;
+  &lt;nodes jvm-gc-options="-XX:+UseConcMarkSweepGC -XX:MaxTenuringThreshold=15 -XX:NewRatio=1 -XX:+PrintGC" jvm-options="-XX:+PrintCommandLineFlags"&gt;
```
Comment (kkraune) [inline documentation/performance/container-tuning.html]: The reference doc is wrong then? https://docs.vespa.ai/documentation/reference/services-container.html#nodes - says jvmargs only - I guess jvm-gc-options is added recently?

File: documentation/performance/container-tuning.html [hunk 2]
```diff
@@ -190,7 +94,7 @@ <h3 id="config-server-and-config-proxy">Config Server and Config Proxy</h3>
 
 
 
-<h2 id="warmup-queries">Warmup queries</h2>
+<h2 id="warmup-queries">Container warmup</h2>
```
Comment (kkraune) [inline documentation/performance/container-tuning.html]: change to id="container-warmup"

File: documentation/performance/container-tuning.html (hunk 1 above)
Comment (jobergum) [reply documentation/performance/container-tuning.html]: it's also in this PR to add those two new options added some time back but not documented.

File: documentation/performance/container-tuning.html (hunk 2 above)
Comment (jobergum) [reply documentation/performance/container-tuning.html]: Done

File: documentation/querying-vespa.html [hunk 3]
```diff
@@ -4,74 +4,79 @@
...
-    When all MLDs have responded to the TLD,
-    it merges the partial results and responds to the search container.
+    The query is sent from the container to <a href="search-definitions.html#searching-multiple-document-types">selected</a> content clusters -
+    see <a href="federation.html">federation</a> for more details. The illustration above only
+   contains one content cluster but multiple is fully supported and allows scaling <a href="search-definitions.html">document types</a> differently. 
+   For example a <em>tweet</em> document type could be indexed in a separate content cluster then <em>user</em> document type which allow for independent scaling. 
   </li><li>
-    The search container performs processing of the results in search chains
-    making further queries if needed before it returns the results to the querying entity.
+    Content clusters have top-level dispatchers (TLDs) which dispatches the query to <a href="content/data-placement.html">content groups</a> within the content cluster.
```
Comment (kkraune) [inline documentation/querying-vespa.html]: as dispatch goes away shortly I would just remove this

Comment (kkraune) [reply documentation/querying-vespa.html]: we will have to remove dispatch from illustrations, too, can do in later PR

File: documentation/querying-vespa.html [hunk 4]
```diff
@@ -4,74 +4,79 @@
...
+    will merge the results and return the merged
+    result up to the calling search container for search phase result processing. 
+    Only the targeted number of <a href="reference/search-api-reference.html#hits">hits</a> is returned.
+    In this phase the only per hit data available is the Vespa internal global document id (gid) 
+    and the relevancy score. In addition there is result meta information like coverage and total hit count. Additional hit
+    specific data like the contents of fields is not available until the result processing phase has completed the content fetching. 
+   </li>
 </ol>
-<p>
-<img src="img/reference/flow/query2.png" />
```
Comment (kkraune) [inline documentation/querying-vespa.html]: this is the only place query2.png is referenced, so remove illustration, too

File: documentation/performance/container-tuning.html (hunk 1 above)
Comment (kkraune) [reply documentation/performance/container-tuning.html]: disregard - I see you fixed this. CREDIT - good job!

File: documentation/querying-vespa.html (hunk 3 above)
Comment (jobergum) [reply documentation/querying-vespa.html]: I think we can leave this until we know how rpc first phase search is going to be enabled 100%

**Pull request merged**